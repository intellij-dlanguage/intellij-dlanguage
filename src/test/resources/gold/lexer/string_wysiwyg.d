import std.stdio;

void main(){
  string a = r"hello";
  string b = r"ab\n"; // string is 4 characters: 'a', 'b', '\', 'n'
  string c = r"c:\games\Sudoku.exe";
  string d = `the Great and Powerful.`;
  string e = `c:\games\Empire.exe`;
  string f = `The "lazy" dog`;
  string g = `a"b\n`;  // string is 5 characters, 'a', '"', 'b', '\', 'n'
  wstring h = r""w;
  dstring i = ``d;
}
