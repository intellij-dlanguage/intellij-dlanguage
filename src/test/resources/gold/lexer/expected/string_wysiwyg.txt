DlangTokenType.import ('import')
WHITE_SPACE (' ')
DlangTokenType.ID ('std')
DlangTokenType.. ('.')
DlangTokenType.ID ('stdio')
DlangTokenType.; (';')
WHITE_SPACE ('\n\n')
DlangTokenType.void ('void')
WHITE_SPACE (' ')
DlangTokenType.ID ('main')
DlangTokenType.( ('(')
DlangTokenType.) (')')
DlangTokenType.{ ('{')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('a')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.WYSIWYG_STRING ('r"hello"')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('b')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.WYSIWYG_STRING ('r"ab\n"')
DlangTokenType.; (';')
WHITE_SPACE (' ')
DlangTokenType.LINE_COMMENT ('// string is 4 characters: 'a', 'b', '\', 'n'')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('c')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.WYSIWYG_STRING ('r"c:\games\Sudoku.exe"')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('d')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.ALTERNATE_WYSIWYG_STRING ('`the Great and Powerful.`')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('e')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.ALTERNATE_WYSIWYG_STRING ('`c:\games\Empire.exe`')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('f')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.ALTERNATE_WYSIWYG_STRING ('`The "lazy" dog`')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('string')
WHITE_SPACE (' ')
DlangTokenType.ID ('g')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.ALTERNATE_WYSIWYG_STRING ('`a"b\n`')
DlangTokenType.; (';')
WHITE_SPACE ('  ')
DlangTokenType.LINE_COMMENT ('// string is 5 characters, 'a', '"', 'b', '\', 'n'')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('wstring')
WHITE_SPACE (' ')
DlangTokenType.ID ('h')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.WYSIWYG_STRING ('r""w')
DlangTokenType.; (';')
WHITE_SPACE ('\n  ')
DlangTokenType.ID ('dstring')
WHITE_SPACE (' ')
DlangTokenType.ID ('i')
WHITE_SPACE (' ')
DlangTokenType.= ('=')
WHITE_SPACE (' ')
DlangTokenType.ALTERNATE_WYSIWYG_STRING ('``d')
DlangTokenType.; (';')
WHITE_SPACE ('\n')
DlangTokenType.} ('}')